### PR TITLE
Remove old temperature and moisture dependance 

### PR DESCRIPTION
**Pull request by jacobcook1995**
# Description

This PR removes a couple of functions that calculate the impact temperature and moisture have on soil processes, as they are not really relevant to the planned soil model. These are replaced by new functions to calculate the impact that temperature and soil water potential on specifically microbial rates.

I decided against removing the old moisture function entirely as this would involve changing how significant parts of the model works (e.g. sorption of organic matter to minerals). This change will instead come as part of a subsequent pull request that will make the model enzyme-explicit.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_rainforest/models/soil/carbon.py [hunk 1]
```diff
@@ -51,46 +56,61 @@ def calculate_soil_carbon_updates(
...
     lmwc_to_maom = calculate_mineral_association(
         soil_c_pool_lmwc,
         soil_c_pool_maom,
         pH,
         bulk_density,
-        moist_temp_scalar,
+        moist_scalar,
         percent_clay,
         constants,
     )
```
Comment (vgro) [inline virtual_rainforest/models/soil/carbon.py:87]: For consistency, could you add the argument names here, too? For example: `moist_scalar=moist_scalar`

File: virtual_rainforest/models/soil/carbon.py [hunk 2]
```diff
@@ -251,48 +269,74 @@ def calculate_binding_coefficient(
...
-    denominator = temp_scalar_coefficient_2 + (
-        temp_scalar_coefficient_3 / np.pi
-    ) * np.arctan(
-        np.pi
-        * temp_scalar_coefficient_4
-        * (temp_scalar_reference_temp - temp_scalar_coefficient_1)
+    return np.exp(
+        (-activation_energy / gas_constant)
+        * ((1 / (soil_temperature + 273.15)) - (1 / (reference_temperature + 273.15)))
     )
```
Comment (vgro) [inline virtual_rainforest/models/soil/carbon.py:299]: I think `273.15` is a good candidate for a universal constant

File: tests/models/soil/test_carbon.py [hunk 3]
```diff
@@ -14,41 +14,48 @@
...
 
 @pytest.fixture
-def moist_temp_scalars(dummy_carbon_data, top_soil_layer_index):
-    """Combined moisture and temperature scalars based on dummy carbon data."""
-    from virtual_rainforest.models.soil.carbon import (
-        convert_moisture_to_scalar,
-        convert_temperature_to_scalar,
-    )
+def moist_scalars(dummy_carbon_data, top_soil_layer_index):
+    """Combined moisture scalars based on dummy carbon data."""
```
[the remark this answers is not in the document]
Comment (jacobcook1995) [reply tests/models/soil/test_carbon.py:18]: This is an artefact from the previous version which combined temperature and moisture scalars into a single scalar.

**Review by vgro — APPROVED** [review]
LGTM. I cannot say much about the science but it all looks sensible to me.

File: virtual_rainforest/models/soil/carbon.py (hunk 2 above)
Comment (davidorme) [reply virtual_rainforest/models/soil/carbon.py:299]: I have added a draft `CoreConsts`  over in #323, so we can start to move things in and work out how to pass them around.

**Review by davidorme — APPROVED** [review]
LGTM

**Pull request merged**